src/timetable_cmd.cpp

```diff
@@ -300,10 +300,10 @@ static bool VehicleTimetableSorter(Vehicle * const &a, Vehicle * const &b)
  * @param flags Operation to perform.
  * @param veh_id Vehicle ID.
  * @param timetable_all Set to set timetable start for all vehicles sharing this order
- * @param start_date The timetable start date.
+ * @param start_ticks Ticks until the timetable starts.
  * @return The error or cost of the operation.
  */
-CommandCost CmdSetTimetableStart(DoCommandFlag flags, VehicleID veh_id, bool timetable_all, TimerGameCalendar::Date start_date)
+CommandCost CmdSetTimetableStart(DoCommandFlag flags, VehicleID veh_id, bool timetable_all, TimerGameTick::Ticks start_ticks)
```

Comment:
I think this means the actual start time varies depending on network client latency. Not sure if that's a problem, but does make it a bit non-deterministic.

Reply:
Is `TimerGameCalendar::day_fract` not synced between clients?

Reply:
Or are you saying, send a start time in ticks but don't apply date_fract until we get inside the command?

Reply:
I'm saying that (I think) sending a relative ticks means it's relative to when the server receives the command, not relative to when the client sends the command. With the current absolute date that isn't the case (because it's absolute.)

Reply:
What about if we revert to sending the date (and eventually a `bool extra_second` once I add seconds mode) in the command call, but then actually store the start date as ticks? Would that be safe?

(Who actually decides when the command runs? The server?)

Reply:
Yes, the server decides. For now you could send as Date and the command can convert to ticks internally.

If there's the concept of absolute "current tick" then you could send current tick and relative tick so that the server can adjust based on

Reply:
Or, if the concept of "absolute ticks" exists, use that...

Reply:
There is an absolute tick counter, but it loops, so probably wouldn't be suitable.

Reply:
TimerGameTick::counter is 64 bits so looping isn't a huge concern.
You could rebase all the values when it's near wrapping if this was really a concern.

Reply:
64 bits is enough for 2^64 / 74 / 365 = 682 trillion years, I don't think wrapping is an issue.

Okay, so actually `timetable_start` could be also be an absolute tick, avoiding logic changes to how that works (no need to decrement every tick.) — it simply changes from Date to ticks. (But if making it relative internally helps you later than that's fine too.)